format/index.md

```diff
@@ -1004,125 +789,176 @@ Content-Type: application/vnd.api+json
+of these examples might be implemented with a foreign key, while the second
+could be implemented with a join table, but the JSON API protocol would be
+the same in both cases.
+
+> Note: A server may choose to delete the underlying resource if a
+relationship is deleted (as a garbage collection measure).
+
+#### Updating To-One Relationships <a href="#crud-updating-to-one-relationships" id="crud-updating-to-one-relationships" class="headerlink"></a>
 
-A `DELETE` request should only succeed if the relationship is currently defined.
+A server **MUST** respond to `PUT` requests to a *to-one relationship URL* as
+described below.
 
-##### Updating To-Many Relationships <a href="#crud-updating-to-many-relationships" id="crud-updating-to-many-relationships" class="headerlink"></a>
+The `PUT` request **MUST** include a top-level member named `data` containing
+one of:
 
-To-many relationships **MAY** optionally be updated with other attributes by
-including them in a `links` object within the document in a `PUT` request.
+* an object with `type` and `id` members corresponding to the related resource
+* `null`, to remove the relationship
 
-For instance, the following `PUT` request performs a complete replacement of the
-`tags` for an article:
+For example, the following request updates the author of an article:
 
 ```text
-PUT /articles/1
+PUT /articles/1/links/author
 Content-Type: application/vnd.api+json
 Accept: application/vnd.api+json
 
 {
-  "articles": {
-    "id": "1",
-    "title": "Rails is a Melting Pot",
-    "links": {
-      "tags": ["2", "3"]
-    }
-  }
+  "data": { "type": "people", "id": "12" }
 }
 ```
 
-In order to remove every member of a to-many relationship, specify an empty
-array (`[]`) as the value of the relationship.
-
-Alternatively, a to-many relationship **MAY** optionally be accessible at its
-relationship URL (see above).
-
-A `PUT` request sent to the URL of a relationship **SHOULD** completely replace
-every member of the relationship. For example:
+And the following request clears the author of the same article:
 
 ```text
-PUT /articles/1/links/tags
+PUT /articles/1/links/author
 Content-Type: application/vnd.api+json
 Accept: application/vnd.api+json
 
 {
-  "tags": ["2", "3"]
+  "data": null
 }
 ```
 
-Replacing a complete set of data is not always appropriate in a distributed
-system which may involve many editors. An alternative is to allow relationships
-to be added and removed individually. This can be done by making fine-grained
-requests to the relationship URL.
+If the relationship is updated successfully then the server **MUST** return
+a `204 No Content` response.
+
+#### Updating To-Many Relationships <a href="#crud-updating-to-many-relationships" id="crud-updating-to-many-relationships" class="headerlink"></a>
+
+A server **MUST** respond to `PUT`, `POST`, and `DELETE` requests to a *to-many
+relationship URL* as described below.
+
```

Comment:
In the spirit of not requiring the client to stop and consider the differences between updating a to-one vs a to-many relationship, would it make sense for "Updating To-One Relationships" to support `POST` and `DELETE` with as equivalent to `PUT`when the existing relationship is `null` (erroring otherwise) and `PUT` to set the relationship to `null`, respectively?